kolibri/core/logger/api.py

```diff
@@ -675,6 +675,36 @@ def _create_attempt(
             **interaction
         )
 
+    def _get_attemptlog(self, session_id, masterylog_id, user, interaction):
+        if "id" in interaction:
+            try:
+                return AttemptLog.objects.get(
+                    id=interaction["id"],
+                    masterylog_id=masterylog_id,
+                    user=user,
+                )
```

Comment:
If we're querying by primary key here, what do the additional filter params do? Are there cases where a record with that id might exist, but with a different user/masterylog?

Reply:
No - it's just a safety measure in case the attempt id was incorrect, or somehow maliciously selected.